src/WebJobs.Script.WebHost/Controllers/AdminController.cs

```diff
@@ -30,7 +30,7 @@ public AdminController(WebScriptHostManager scriptHostManager)
 
         [HttpPost]
         [Route("admin/functions/{name}")]
-        public HttpResponseMessage Invoke(string name, [FromBody] FunctionInvocation invocation)
+        public async Task<HttpResponseMessage> Invoke(string name, [FromBody] FunctionInvocation invocation, bool block = false)
```

Comment:
How about adding a "WaitForCompletion" property to FunctionInvocation? That was the intent of that type - to contain all the various invocation flags/data.